sys/fido2/ctap/Kconfig

```
@@ -4,13 +4,31 @@
 # General Public License v2.1. See the file LICENSE in the top level
 # directory for more details.
 
-menuconfig KCONFIG_USEMODULE_FIDO2_CTAP
+menuconfig MODULE_FIDO2_CTAP
     bool "FIDO2 CTAP"
-    depends on USEMODULE_FIDO2_CTAP
+    depends on HAS_PERIPH_GPIO
+    depends on HAS_PERIPH_GPIO_IRQ
+    depends on TEST_KCONFIG
+    select PACKAGE_TINYCBOR
+    select PACKAGE_MICRO-ECC
+    select PACKAGE_TINY-ASN1
+    select MODULE_PERIPH_GPIO
+    select MODULE_PERIPH_GPIO_IRQ
+    select MODULE_EVENT
+    select MODULE_EVENT_TIMEOUT
+    select MODULE_ZTIMER
+    select MODULE_ZTIMER_MSEC
+    select MODULE_MTD
+    select MODULE_MTD_FLASHPAGE
+    select MODULE_MTD_WRITE_PAGE
+    select MODULE_RANDOM
+    select MODULE_CRYPTO_AES_256
+    select MODULE_CIPHER_MODES
+    select MODULE_HASHES
```

Missing SHA-256, and it seems to select `MODULE_HASHES` so no need for that.
```suggestion
    select MODULE_PRNG_SHA256PRNG
```

---

Chaning this results in the following error for me:

```
warning: the choice symbol MODULE_PRNG_SHA256PRNG (defined at /home/b0re42/Uni/BachelorArbeit/RIOT/sys/random/Kconfig:56) is selected by the following symbols, but select/imply has no effect on choice symbols
```

---

Ah yes, I did not notice that this is one particular implementation of random. Why do you need an specific implementation of the random API?

---

I don't quite understand what you mean ? I need `MODULE_RANDOM` because `PACKAGE_MICRO-ECC` requires it

---

> I don't quite understand what you mean ? I need `MODULE_RANDOM` because `PACKAGE_MICRO-ECC` requires it

Yes enabling `MODULE_RANDOM` is fine, but you are also trying to select a particular implementation of it (`MODULE_PRNG_SHA256PRNG`).

---

Sorry I still don't understand. You suggested to remove `select MODULE_HASHES` and add `select MODULE_PRNG_SHA256PRNG` instead. This caused errors. So my initial approach was correct?

---

> Sorry I still don't understand. You suggested to remove `select MODULE_HASHES` and add `select MODULE_PRNG_SHA256PRNG` instead. This caused errors. So my initial approach was correct?

Ah ok, I understand. I'm sorry about the confusion. Yeah actually my initial proposal of adding `MODULE_PRNG_SHA256PRNG` was because [it is being done in the `Makefile.dep`](https://github.com/RIOT-OS/RIOT/blob/cbe9d6967b9341362bbeab836ee9b6629d50dc40/sys/Makefile.dep#L846), but now that I checked the symbols in Kconfig I think we should not bring an specific random implementation. So that should be removed in the Makefile.dep. If FIDO requires `MODULE_HASHES` directly then add it back.

---

ahhh ok I see. Removed `prng_sha256prng` form Makefile.dep :)